test/components/dragBoxLayerTests.ts

```diff
@@ -2,284 +2,272 @@
+        dbl = new Plottable.Components.DragBoxLayer();
+        quarterPoint = {
+          x: SVG_WIDTH / 4,
+          y: SVG_HEIGHT / 4
+        };
+        halfPoint = {
+          x: SVG_WIDTH / 2,
+          y: SVG_HEIGHT / 2
+        };
+      });
 
-      let targetPoint = {
-        x: SVG_WIDTH / 2,
-        y: SVG_HEIGHT / 2
-      };
+      it("draws box on drag", () => {
+        dbl.renderTo(svg);
+        assert.isFalse(dbl.boxVisible(), "box is hidden initially");
 
-      let target = dbl.background();
-      TestMethods.triggerFakeDragSequence(target, targetPoint, targetPoint);
+        let target = dbl.background();
+        TestMethods.triggerFakeDragSequence(target, quarterPoint, halfPoint);
+        assert.isTrue(dbl.boxVisible(), "box is drawn on drag");
+        let bounds = dbl.bounds();
+        assert.deepEqual(bounds.topLeft, quarterPoint, "top-left point was set correctly");
+        assert.deepEqual(bounds.bottomRight, halfPoint, "bottom-right point was set correctly");
+        svg.remove();
+      });
 
-      assert.isFalse(dbl.boxVisible(), "box is hidden on click");
+      it("dismisses on click", () => {
+        dbl.renderTo(svg);
 
-      svg.remove();
-    });
+        let target = dbl.background();
 
-    it("clipPath enabled", () => {
-      let svg = TestMethods.generateSVG(SVG_WIDTH, SVG_HEIGHT);
-      let dbl = new Plottable.Components.DragBoxLayer();
-      dbl.renderTo(svg);
-      TestMethods.verifyClipPath(dbl);
-      let clipRect = (<any> dbl)._boxContainer.select(".clip-rect");
-      assert.strictEqual(TestMethods.numAttr(clipRect, "width"), SVG_WIDTH, "the clipRect has an appropriate width");
-      assert.strictEqual(TestMethods.numAttr(clipRect, "height"), SVG_HEIGHT, "the clipRect has an appropriate height");
-      svg.remove();
-    });
+        assert.isFalse(dbl.boxVisible(), "box is hidden initially");
+        TestMethods.triggerFakeDragSequence(target, quarterPoint, halfPoint);
+        assert.isTrue(dbl.boxVisible(), "box is drawn on drag");
+
+        TestMethods.triggerFakeDragSequence(target, halfPoint, halfPoint);
+        assert.isFalse(dbl.boxVisible(), "box is hidden on click");
+        svg.remove();
+      });
+
+      it("generates the correct clipPath", () => {
+        dbl.renderTo(svg);
+
+        TestMethods.verifyClipPath(dbl);
+        let clipRect = (<any> dbl)._boxContainer.select(".clip-rect");
+        assert.strictEqual(TestMethods.numAttr(clipRect, "width"), SVG_WIDTH, "the clipRect has an appropriate width");
+        assert.strictEqual(TestMethods.numAttr(clipRect, "height"), SVG_HEIGHT, "the clipRect has an appropriate height");
+        svg.remove();
+      });
+
+      it("can get and set the detection radius", () => {
+        assert.strictEqual(dbl.detectionRadius(), 3, "there is a default detection radius");
+        assert.doesNotThrow(() => dbl.detectionRadius(4), Error, "can set detection radius before anchoring");
```

Comment:
`doesNotThrow()`. I got tricked by this 3 times while refactoring. The signature is a bit weird.